Improve Sophia documentation with Argo/GPT4o (#564)
argonne-lcf/user-guides: docs/sophia/compiling-and-linking/compiling-and-linking-overview.md

# Compiling and Linking on Sophia

## Overview

> Sophia has AMD processors on the login nodes (sophia-login-01,02) and AMD processors and NVIDIA A100 GPUs on the compute nodes (see [Machine Overview](../hardware-overview/machine-overview.md) page). The login nodes can be used to create containers and launch jobs.

Reviewer: ```suggestion
Sophia has AMD processors on the login nodes (`sophia-login-01,02`) and AMD processors and NVIDIA A100 GPUs on the compute nodes (see [Machine Overview](../hardware-overview/machine-overview.md) page). The login nodes can be used to create containers and launch jobs.
```

> **Note:** Until the cross-compiling environment is set up or dedicated build nodes are added, the compute nodes will have to be used for compiling. Do not compile codes on the login nodes. To launch an interactive job and acquire a compute node for compiling, use:

Reviewer: ```suggestion
!!! warning inline end "Must compile on a compute node" 
    
    Until the cross-compiling environment is set up or dedicated build nodes are added, the compute nodes will have to be used for compiling. Do not compile codes on the login nodes. To launch an interactive job and acquire a compute node for compiling, use:
```

> ```bash
> qsub -I -q workq -A myProjectShortName -n 1 -t HH:MM:SS
> ```

Reviewer: ```suggestion
    ```bash
    qsub -I -q workq -A myProjectShortName -n 1 -t HH:MM:SS
    ```
```

nodes get added, the compute nodes will have to be used for compiling. Do not
compile codes on the login nodes. To launch an interactive job and acquire a
compute node for compiling, use
**Note:** Until the cross-compiling environment is set up or dedicated build nodes are added, the compute nodes will have to be used for compiling. Do not compile codes on the login nodes. To launch an interactive job and acquire a compute node for compiling, use:

```
```bash
qsub -I -q workq -A myProjectShortName -n 1 -t HH:MM:SS
```

> The default programming environment on the Sophia compute nodes is the GNU compiler tools coupled with NVIDIA’s CUDA toolkit. 

Reviewer: Argo really doesnt like trailing whitespace, which I agree with
